hooks/tk-multi-publish2/ingest/ingest_cdl_files.py

```diff
@@ -0,0 +1,246 @@
+        finalize methods.
+
+        A dictionary on the following form::
+
+            {
+                "Settings Name": {
+                    "type": "settings_type",
+                    "default_value": "default_value",
+                    "description": "One line description of the setting"
+            }
+
+        The type string should be one of the data types that toolkit accepts
+        as part of its environment configuration.
+        """
+        schema = super(IngestCDLFilesPlugin, self).settings_schema
+        schema["Item Type Filters"]["default_value"] = ["file.cdl"]
+        return schema
+
+    @property
+    def description(self):
+        """
+        Verbose, multi-line description of what the plugin does. This can
+        contain simple html for formatting.
+        """
+
+        return """
+        Plugin to process CDL files, once the files are processed it publishes an avid_grade.cc, after validating that
+        'Slope', 'Offset', 'Power', 'Saturation' fields are intact in the input file.
+        """
+
+    def accept(self, task_settings, item):
+        """
+        Method called by the publisher to determine if an item is of any
+        interest to this plugin. Only items matching the filters defined via the
+        item_filters property will be presented to this method.
+
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        # Run the parent acceptance method
+        accept_data = super(IngestCDLFilesPlugin, self).accept(task_settings, item)
+
+        if item.type == "file.cdl":
+            accept_data["accepted"] = True
+        else:
+            accept_data["accepted"] = False
+
+        return accept_data
+
+    def validate(self, task_settings, item):
+        """
+        Validates the given item to check that it is ok to publish.
+
+        Returns a boolean to indicate validity.
+
+        :param task_settings: Dictionary of settings
+        :param item: Item to process
+
+        :returns: True if item is valid, False otherwise.
+        """
+
+        fields = item.properties["fields"]
+
+        # to override the name of the published file
+        if fields["name"] != "avid_grade":
+            fields["name"] = "avid_grade"
```

Comment:
This seems wonky. Can you explain why this is necessary?

Reply:
Our OCIO config setup is supposed to read avid_grade.cc files, since we get these(ccc or cc) files from the client with different names, the name field resolves to a different value and we are simply correcting it here, before publish.

`shot_publish_cdl:                       '@shot_publish_area_cdl/grades/grade_v{version}/{name}.cc'`